docs/source/frameworks/fastai.rst

```diff
@@ -0,0 +1,260 @@
+
+To quickly save the trained learner, use ``save_model``:
+
+.. code-block:: python
+
+   bentoml.fastai.save_model("fastai_sentiment", learner)
+
+.. code-block:: bash
+
+   Model(tag="fastai_sentiment:5bakmghqpk4z3gxi", path="~/bentoml/models/fastai_sentiment/5bakmghqpk4z3gxi/")
+
+In addition to :ref:`general options <concepts/model:Save A Trained Model>`
+provided by :code:`save_model`, you can optionally provide a different ``pickle_module``
+for serializing the model.
+
+.. code-block:: python
+
+   import pickle
+   
+   bentoml.fastai.save_model("fastai_sentiment", learner, pickle_module=pickle)
+
+.. note::
+
+   By default, BentoML uses ``cloudpickle`` for serialization. The key difference from Python's ``pickle`` is that 
+   ``cloudpickle`` has the capability to serialize functions and so it can directly serialize members of the object without reference to its type.
+
+   :bdg-primary:`Our Recommendation:` ``cloudpickle`` *should be used for most cases.*
+
+.. admonition:: about :code:`save_model()` behaviour
+
+   BentoML also tries to remove some known callback that have issues with
+   serialization, such as ``ParamScheduler``. 
+
+   We found that some of given callback are only useful during training phase, not during serving, thus 
+   not affecting inference results.
+
+
+Loading a learner with BentoML
+------------------------------
+
+To load the learner back to memory, use ``load_model``:
+
+.. code-block:: python
+
+   learner = bentoml.fastai.load_model("fastai_sentiment")
+
+Proceed to then test the learner with prediction inputs:
+
+.. code-block:: python
+
+   learner.predict("I really liked that movie!")
+
+In addition to :ref:`general options <concepts/model:Retrieve a saved model>`
+provided by :code:`load_model`, you can also provide ``cpu`` to enforce loading
+the learner on CPU.
+
+.. code-block:: python
+
+   learner = bentoml.fastai.load_model("fastai_sentiment", cpu=False)
+
+
+.. admonition:: About the behaviour of :code:`cpu=True`
+
+   **fastai** will determine which devices to use (GPU or CPU) via ``cpu``. The
+   results will then be passed down to ``map_location`` of ``torch.load``.
+   Refers to `PyTorch's documentation <https://pytorch.org/docs/stable/generated/torch.load.html#torch-load>`_
+   for more information.
+
+   Additionallly, if the model uses ``mixed_precision``, then the loaded model will also be converted to FP32.
+   Learn more about `mixed precision <https://docs.fast.ai/callback.fp16.html>`_.
+
+
+   :bdg-primary:`Remarks:` BentoML are currently only providing CPU supports for **fastai**.
+
+
+Using Runners
+-------------
+
+.. seealso::
+
+   :ref:`Runners' documentation<concepts/runner:Using Runners>` on Runners' concept and its usage.
+
+.. seealso::
+
+   :ref:`Specifying Runner Resources<concepts/runner:Specifying Required Resources>` on providing options for Runners.
+
+
+To use ``fastai`` runner locally, access the model via ``get`` and convert it to
```

Comment:
I feel like the distinction between `get` and `load_model` isn't very clear, but then again that may just be difficult to get across in general.

I'm somewhat of the opinion the guide shouldn't touch ``load_model`` at all, that shouldn't normally be part of the workflow, right? Maybe we can have a subsection or something which explains users can use it if they need access to the `Learner` or something instead of the full section above.